pkg/operator/controllers/previewfeature/nsgflowlogs/nsgflowlogs.go

```diff
@@ -0,0 +1,139 @@
+package nsgflowlogs
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	aropreviewv1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/preview.aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/util/subnet"
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-08-01/network"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/sirupsen/logrus"
+	"time"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/network"
+)
+
+func NewFeature(flowLogsClient network.FlowLogsClient, kubeSubnets subnet.KubeManager, subnets subnet.Manager, log *logrus.Entry) *nsgFlowLogsFeature {
+	return &nsgFlowLogsFeature{
+		log:            log,
+		kubeSubnets:    kubeSubnets,
+		flowLogsClient: flowLogsClient,
+		subnets:        subnets,
+	}
+}
+
+type nsgFlowLogsFeature struct {
+	log *logrus.Entry
+
+	kubeSubnets    subnet.KubeManager
+	subnets        subnet.Manager
+	flowLogsClient network.FlowLogsClient
+}
+
+func (n *nsgFlowLogsFeature) Name() string {
+	return "nsgFlowLogsFeature"
+}
+
+func (n *nsgFlowLogsFeature) Reconcile(ctx context.Context, instance *aropreviewv1alpha1.PreviewFeature) error {
+	if instance.Spec.NSGFlowLogs == nil {
+		return nil
+	}
+
+	if !instance.Spec.NSGFlowLogs.Enabled {
+		return n.Disable(ctx, instance)
+	}
+
+	return n.Enable(ctx, instance)
+}
+
+func (n *nsgFlowLogsFeature) Enable(ctx context.Context, instance *aropreviewv1alpha1.PreviewFeature) error {
+	// build a request as described here https://docs.microsoft.com/en-us/azure/network-watcher/network-watcher-nsg-flow-logging-rest#enable-network-security-group-flow-logs
+	flowLog := mgmtnetwork.FlowLog{
+		Location: to.StringPtr(instance.Spec.NSGFlowLogs.Location),
+		FlowLogPropertiesFormat: &mgmtnetwork.FlowLogPropertiesFormat{
+			Enabled: to.BoolPtr(true),
+			Format: &mgmtnetwork.FlowLogFormatParameters{
+				Type:    mgmtnetwork.JSON,
+				Version: to.Int32Ptr(int32(instance.Spec.NSGFlowLogs.Version)),
+			},
+			RetentionPolicy: &mgmtnetwork.RetentionPolicyParameters{
+				Days: to.Int32Ptr(instance.Spec.NSGFlowLogs.RetentionDays),
+			},
+			StorageID: to.StringPtr(instance.Spec.NSGFlowLogs.StorageAccountResourceID),
+			FlowAnalyticsConfiguration: &mgmtnetwork.TrafficAnalyticsProperties{
+				NetworkWatcherFlowAnalyticsConfiguration: &mgmtnetwork.TrafficAnalyticsConfigurationProperties{
+					WorkspaceID:              to.StringPtr(instance.Spec.NSGFlowLogs.TrafficAnalyticsLogAnalyticsWorkspaceID),
+					TrafficAnalyticsInterval: to.Int32Ptr(int32(instance.Spec.NSGFlowLogs.TrafficAnalyticsInterval.Truncate(time.Minute).Minutes())),
+				},
+			},
+		},
+	}
+
+	nsgs, err := n.getNSGs(ctx)
+	if err != nil {
+		return err
+	}
+
+	for nsgID, _ := range nsgs {
+		flowLog.FlowLogPropertiesFormat.TargetResourceID = to.StringPtr(nsgID)
+		NetworkWatcherResource, err := azure.ParseResourceID(instance.Spec.NSGFlowLogs.NetworkWatcherID)
```

Comment:
Nitpicking on code style :)


```suggestion
		networkWatcherResource, err := azure.ParseResourceID(instance.Spec.NSGFlowLogs.NetworkWatcherID)
```